format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingConfig.java

```diff
@@ -141,15 +147,18 @@ public Path getFilePathForSchemaGeneration(String path, String regexPathFilter,
 
     for (FileStatus file : files) {
       if (Strings.isNullOrEmpty(regexPathFilter)) {
-        return file.getPath();
+        paths.add(file.getPath());
       } else {
         Pattern pattern = Pattern.compile(regexPathFilter);
```

Comment:
Don't compile pattern inside a loop. Compile it outside and use matcher inside the loop.